core/src/operations/mod.rs

```diff
@@ -355,109 +355,108 @@ impl Operation {
     pub fn op_code(&self) -> u8 {
         match self {
-            Self::Noop => 0,
-            Self::Assert => 1,
-
-            Self::FmpAdd => 2,
-            Self::FmpUpdate => 3,
-
-            Self::Push(_) => 4,
-
-            Self::Eq => 0b0100_1001,
-            Self::Eqz => 5,
-
-            Self::Add => 0b0100_1000,
-            Self::Neg => 7,
-            Self::Mul => 0b0100_1010,
-            Self::Inv => 8,
-            Self::Incr => 9,
-            Self::And => 0b0100_1011,
-            Self::Or => 0b0100_1100,
-            Self::Not => 11,
-
-            Self::Pad => 12,
-            Self::Drop => 13,
-
-            Self::Dup0 => 14,
-            Self::Dup1 => 15,
-            Self::Dup2 => 16,
-            Self::Dup3 => 17,
-            Self::Dup4 => 18,
-            Self::Dup5 => 19,
-            Self::Dup6 => 20,
-            Self::Dup7 => 21,
-            Self::Dup9 => 22,
-            Self::Dup11 => 23,
-            Self::Dup13 => 24,
-            Self::Dup15 => 25,
-
-            Self::Swap => 26,
-            Self::SwapW => 27,
-            Self::SwapW2 => 28,
-            Self::SwapW3 => 29,
-            Self::SwapDW => 30,
-
-            Self::MovUp2 => 31,
-            Self::MovUp3 => 32,
-            Self::MovUp4 => 33,
-            Self::MovUp5 => 34,
-            Self::MovUp6 => 35,
-            Self::MovUp7 => 36,
-            Self::MovUp8 => 37,
-
-            Self::MovDn2 => 40,
-            Self::MovDn3 => 41,
-            Self::MovDn4 => 42,
-            Self::MovDn5 => 43,
-            Self::MovDn6 => 44,
-            Self::MovDn7 => 45,
-            Self::MovDn8 => 46,
-
-            Self::CSwap => 50,
-            Self::CSwapW => 51,
-
+            Self::Noop => 0b0000_0000,
+            Self::Eqz => 0b0000_0001,
+            Self::Neg => 0b0000_0010,
+            Self::Inv => 0b0000_0011,
+            Self::Incr => 0b0000_0100,
+            Self::Not => 0b0000_0101,
+            Self::FmpAdd => 0b0000_0110,
+            Self::MLoad => 0b0000_0111,
+            Self::Swap => 0b0000_1000,
+            // empty: 0b0000_1001
+            Self::MovUp2 => 0b0000_1010,
+            Self::MovDn2 => 0b0000_1011,
+            Self::MovUp3 => 0b0000_1100,
+            Self::MovDn3 => 0b0000_1101,
+            Self::ReadW => 0b0000_1110,
+            // empty: 0b0000_1111
+            Self::MovUp4 => 0b0001_0000,
+            Self::MovDn4 => 0b0001_0001,
+            Self::MovUp5 => 0b0001_0010,
+            Self::MovDn5 => 0b0001_0011,
+            Self::MovUp6 => 0b0001_0100,
+            Self::MovDn6 => 0b0001_0101,
+            Self::MovUp7 => 0b0001_0110,
+            Self::MovDn7 => 0b0001_0111,
+            Self::SwapW => 0b0001_1000,
+            // empty: 0b0001_1001
+            Self::MovUp8 => 0b0001_1010,
+            Self::MovDn8 => 0b0001_1011,
+            Self::SwapW2 => 0b0001_1100,
+            Self::SwapW3 => 0b0001_1101,
+            Self::SwapDW => 0b0001_1110,
```

Comment:
These don't match what's listed in the issue, since `SwapW` has been moved earlier. I liked the logic of the previous ordering, so I'm curious - what the reasoning is for this change? (Also, I just want to make sure the spreadsheet isn't out of sync, in case it's going to get added to the docs)

Reply:
I've updated the spreadsheet after I took the screenshot - so, there are a couple of minor difference between what is in the issue and what is in the spreadsheet/code.

The reason for this re-arrangement is that I wanted to put operations which affect the first two words of the stack (i.e., up to element $7$) before operations which affect the 3rd and the 4th word. Originally I made a mistake thinking that `MOVUP8` and `MOVDN8` affect only the first two words, but then realized that they affect the first element of the 3rd word, and thus, moved them after `SWAPW` (which affects only the first two words).

We may re-evaluate this approach in the future though. In general, I think rearranging operation order within a given group should be fairly straightforward.

Reply:
Sure, this makes sense. Thanks for explaining!